Headers/Foundation/NSURL.h

```diff
@@ -85,29 +85,29 @@ enum
  * Calls -initFileURLWithPath: which escapes characters in the
  * path where necessary.
  */
-+ (id) fileURLWithPath: (NSString*)aPath;
++ (instancetype) fileURLWithPath: (NSString*)aPath;
```

Comment:
Interestingly Apples uses (NSURL *) for all the factory methods.

Reply:
I noticed that too, but it seemed more like a case of API not having been updated.

That being said, please let me know if you prefer matching Apple’s headers.

Reply:
I guess Apple uses NSURL rather than instancetype because there are no subclasses (and perhaps they don't intend NSURL to be subclassed), so every returned instance must be an NSURL.  I have no strong feelings about which is better (both are better than id).